File: Rover/Direction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Rover
{
    public enum Direction
    {
        East=0,
        NorthEast=45,
        North=90,
        NorthWest=135,
        West=180,
        SouthWest=225,
        South=270,
        SouthEast=315,
        OtherEast=360,
        otherSouthEast=-45
    }
}

File: src/main/java/com/tavisca/workshops/pratham/rover/model/Direction.java
package com.tavisca.workshops.pratham.rover.model;

public enum Direction {
    NORTH, EAST, SOUTH, WEST;

    private static Direction directionArray[] = values();

    public Direction right() {
        return directionArray[mod(this.ordinal() + 1, directionArray.length)];
    }

    public Direction left() {
        return directionArray[mod(this.ordinal() - 1, directionArray.length)];
    }

    private int mod(int a, int b) {
        return ((a % b) + b) % b;
    }
}
